executor/distsql.go

```diff
@@ -302,6 +303,24 @@ func (e *IndexReaderExecutor) open(ctx context.Context, kvRanges []kv.KeyRange)
 			return err
 		}
 	}
+	// Global index needs partitionIds to be pushed down for selected pids' result.
+	if e.index.Global {
+		visiblePartitionIDs := e.table.GetPartitionedTable().GetAllPartitionIDs()
+		visiblePartitionIDMap := make(map[int64]struct{}, len(visiblePartitionIDs))
+		for i := range visiblePartitionIDs {
+			vpid := visiblePartitionIDs[i]
+			visiblePartitionIDMap[vpid] = struct{}{}
+		}
+
+		idxScanExec := e.dagPB.Executors[0]
+		idxScanExec.IdxScan.IsGlobal = &e.index.Global
+		idxScanExec.IdxScan.PartitionIds = make([]int64, 0, len(e.partitionIDMap))
+		for pid := range e.partitionIDMap {
+			if _, exist := visiblePartitionIDMap[pid]; exist {
+				idxScanExec.IdxScan.PartitionIds = append(idxScanExec.IdxScan.PartitionIds, pid)
+			}
+		}
```

Comment:
Could this be a helper function and used both here and in `(e *IndexLookUpExecutor) open()` to avoid code duplication?

Reply:
resolved